packages/runner/src/providers/container/index.ts

```diff
@@ -0,0 +1,213 @@
+import { Docker } from "node-docker-api"
+import fetch from 'node-fetch'
+import { ReadStream } from 'fs'
+import API from '@mesg/api/lib/lcd'
+import * as ServiceType from '@mesg/api/lib/typedef/service';
+import { IService } from "@mesg/api/lib/service-lcd"
+import { IRunner } from '@mesg/api/lib/runner-lcd'
+import { findHash } from "@mesg/api/lib/util/txevent"
+import { Provider } from '../../index'
+import { Network } from "node-docker-api/lib/network"
+import Container from './container'
+import { createHash } from 'crypto'
+import { IAccount } from "@mesg/api/lib/account-lcd";
+
+const debug = require('debug')('runner')
+
+const ENGINE_NETWORK_NAME = 'engine'
+
+type Labels = { [key: string]: string }
+
+export default class DockerContainer implements Provider {
+
+  private _api: API
+  private _client: Docker
+  private _mnemonic: string
+  private ipfsGateway = "http://ipfs.app.mesg.com:8080/ipfs"
+  private serviceEndpoint: string
+
+  constructor(endpoint: string, mnemonic: string, serviceEndpoint: string = ENGINE_NETWORK_NAME) {
+    this._api = new API(endpoint)
+    this._client = new Docker(null)
+    this._mnemonic = mnemonic
+    this.serviceEndpoint = serviceEndpoint
+  }
+
+  async start(serviceHash: string, env: string[]): Promise<IRunner> {
+
+    debug('fetch service')
+    const service = await this._api.service.get(serviceHash)
+    const account = await this._api.account.import(this._mnemonic)
+    const { runnerHash, instanceHash, envHash } = await this._api.runner.hash(account.address, service.hash, env)
+    debug('broadcast tx')
+
+    const labels = {
+      'mesg.service': service.hash,
+      'mesg.runner': runnerHash,
+    }
+
+    const image = await this.build(service)
+
+    let serviceNetwork = (await this.findNetwork(labels) as Network)
+    if (!serviceNetwork) serviceNetwork = await this._client.network.create({ name: service.hash, labels })
```

Comment:
The service's network is recreated every time I use the command `service:dev`. I end up with duplication:
```
➜  js-sdk git:(feature/runner-lib) docker network list
NETWORK ID          NAME                                           DRIVER              SCOPE
b86a7769deef        ERVPJwMwZcDoJv5em8nxVAvPRqAXf2XiTeGLeToMbYHh   bridge              local
c0fe50d14a3f        ERVPJwMwZcDoJv5em8nxVAvPRqAXf2XiTeGLeToMbYHh   bridge              local
af8a376d624d        ERVPJwMwZcDoJv5em8nxVAvPRqAXf2XiTeGLeToMbYHh   bridge              local
92940aa1e03b        ERVPJwMwZcDoJv5em8nxVAvPRqAXf2XiTeGLeToMbYHh   bridge              local
6c2c6a5f1736        bridge                                         bridge              local
b13df5e90b95        docker_gwbridge                                bridge              local
feb53a7e53db        engine                                         bridge              local
213a022fe26a        host                                           host                local
4u5mrrqsqv6p        ingress                                        overlay             swarm
1214a3f2b782        none                                           null                local
```
The network is also not deleted when the service is stopped.

Reply:
Could be nice also to prefix the name of the network with `mesg_srv_`

Reply:
done